### PR TITLE
add nipals algorithm

### PR DESCRIPTION
**Pull request by jobo322**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
   },
   "homepage": "https://github.com/mljs/pca",
   "dependencies": {
+    "jest": "^22.1.4",
```
Comment (targos) [inline package.json]: Can you remove it? We still use mocha in this project

**Comment by lpatiny** [comment]
Both algorithms should not yield to similar results ?

**Comment by targos** [comment]
@lpatiny see https://cran.r-project.org/web/packages/nipals/vignettes/nipals_algorithm.pdf

**Comment by lpatiny** [comment]
@jobo322 
* could you migrate the test to jest in one commit ?
* could you confirm that both algorithms give the same results ?
* could you give an example with missing values and retrieving just the firsts principal components or all of them ?
* could you split the file and one file called 'nipals.js' ?
* it seems the documentation was not updated and I guess there should be a new parameter to tell how many components you want to retrieve

File: package.json [hunk 2]
```diff
@@ -37,14 +43,17 @@
   },
   "homepage": "https://github.com/mljs/pca",
   "dependencies": {
+    "jest-matcher-deep-close-to": "^1.3.0",
```
Comment (targos) [inline package.json]: should be in devDependencies

**Comment by targos** [comment]
this was done in Matrix

**Pull request closed without merging**